src/os/bluestore/BlueStore.cc

```diff
@@ -11965,6 +11977,63 @@ void BlueStore::_kv_finalize_thread()
   kv_finalize_started = false;
 }
 
+void BlueStore::_zoned_cleaner_start() {
+  dout(10) << __func__ << dendl;
+
+  zoned_cleaner_thread.create("bstore_zcleaner");
+}
+
+void BlueStore::_zoned_cleaner_stop() {
+  dout(10) << __func__ << dendl;
+  {
+    std::unique_lock l{zoned_cleaner_lock};
+    while (!zoned_cleaner_started) {
```

Comment:
You might want to check zoned_cleaner_stopped for true at this point and bypass everything till join() call.. Just to be able to handle multiple _zoned_cleaner_stop() requests properly.
This requires 'true' to be preserved in zoned_cleaner_stopped till the next thread start if any.
Certainly not a big deal for now -  just makes the code a bit more common.

Reply:
So when implementing this, I tried to preserve the structure of _kv_sync_thread() and kv_stop().  Unless you feel strongly about this, I'd rather preserve the same structure as the existing code because I think that one is more common than what you suggest.

Reply:
OK, makes sense